symbolic-unwind/src/base.rs

```diff
@@ -0,0 +1,182 @@
+//! Basic definitions necessary for stack unwinding.
+use std::convert::TryInto;
+use std::fmt::Debug;
+use std::ops::{Add, Div, Mul, Rem, Sub};
+
+/// Trait that abstracts over the [endianness](https://en.wikipedia.org/wiki/Endianness)
+/// of data representation.
+///
+/// This trait provides no other functionality than a method for testing whether
+/// an endianness is big or little. In particular it does not provide methods for
+/// reading number types the way that similar traits/types in `byteorder` and `gimli` do.
```

Comment:
I wonder if you could extend this to very briefly describe why defining our own is preferable over re-using those?  That'll probably be interesting reading for anyone reading this code in the future.